src/main/java/com/flowpowered/nbt/stream/NBTInputStream.java

```diff
@@ -54,6 +56,22 @@
  * may be found at <a href="https://flowpowered.com/nbt/spec.txt"> https://flowpowered.com/nbt/spec.txt</a>.
  */
 public final class NBTInputStream implements Closeable {
+
+	/**
+	 * Flag indicating that the given data stream is not compressed.
+	 */
+	public static final int NO_COMPRESSION = 0;
+	/**
+	 * Flag indicating that the given data will be compressed with the GZIP compression algorithm. This is the default compression method used to compress nbt files.
+	 * Chunks in Minecraft Region/Anvil files with compression method {@code 1} (see the respective format documentation) will use this compression method too, although this is not actively used anymore.
+	 */
+	public static final int GZIP_COMPRESSION = 1;
+	/**
+	 * Flag indicating that the given data will be compressed with the ZLIB compression algorithm. This is the default compression method used to compress the nbt
+	 * data of the chunks in Minecraft Region/Anvil files, but only if its compression method is {@code 2} (see the respective format documentation), which is default for all newer versions.
+	 */
+	public static final int ZLIB_COMPRESSION = 2;
```

Comment:
You did all this work and added all this extra code just to support a compression method that only region files use? o.O
I would just decompress the data in your region file abstraction and feed uncompressed data to NBTInputStream

Reply:
There are three ways chunk data can be compressed. And if the `NBTInputStream` provides syntactic sugar to unwrap one compression, I thought that I should add the other one as well (the first time I tried to load region files I didn't know they had custom compression so I thought there was a bug or so, quite painful).
I do agree that this was kind of a poor design decision (the library should not handle compression/endianness at all), but I wanted this change to be backwards-compatible.

Reply:
One could just require the user to instantiate EndianSwitchableInputStream and make NBTInputStream's constructor require a DataInput instead of InputStream. I would still have EndianSwitchableInputStream in the library. The current helper constructors that support endianness internally are not that bad though... If its not broken don't fix it.

You could create a CompressedInputStream class that supports all three compression types. Wait why would you do that? The only methods common to all three types are already in InputStream. See where this is going?